sdks/go/container/boot.go

```diff
@@ -110,12 +111,23 @@ func main() {
 	default:
 		found := false
 		for _, a := range artifacts {
-			n, _ := artifact.MustExtractFilePayload(a)
-			if n == worker {
+			if a.GetRoleUrn() == "beam:artifact:role:go_worker_binary:v1" {
```

Comment:
adding to https://github.com/apache/beam/blob/1da7bb5b411c7e2854ad68a4c320594312461681/sdks/go/pkg/beam/artifact/materialize.go#L45 instead of using string literal?